### PR TITLE
Do not return Response<T> for service methods that may not call the service

### PR DESCRIPTION
**Pull request by srnagar**
This guideline is based on the existing patterns used in Key Vault and Identity:

- [CryptographyClient](https://apiview.dev/Assemblies/Review/3657b535be7a41d0952418b69d33f880#com.azure.security.keyvault.keys.cryptography.CryptographyAsyncClient) in KeyVault - here service methods compute the response locally if the key is locally available or calls the service if it's not. So, `WithResponse` methods are not offered for most methods. 
- All credential types in Azure Identity do not offer `WithResponse` methods.

### CODE REVIEW

File: docs/java/introduction.md [hunk 1]
```diff
@@ -453,6 +453,10 @@ For methods that combine multiple requests into a single call:
 
 {% include requirement/MUST id="java-response-errors" %} provide enough information in failure cases for a developer to take appropriate corrective action, including a message describing what went wrong and details on the corrective actions to take.
 
+For methods that may not make a service call each time it is invoked either because the response is cached by the client or the response can be computed locally:
+
+{% include requirement/MUSTNOT id="java-no-withresponse" %} offer `WithResponse` service method that returns `Response<T>`. All overloads should return `T` instead.
```
Comment (JonathanGiles) [inline docs/java/introduction.md:458]: This is a little unclear to me. I think you're saying you must not offer any 'withResponse' method, but then the mention of 'all overloads' throws me off. Wouldn't the guidance be to not offer a WithResponse method at all for these types?

**Comment by weshaggard** [comment]
@srnagar what is the next step to getting this merged?

**Comment by vhvb1989** [comment]
@srnagar , just for the record, on C++ we decided that the SDK client should not provide `local-only` operations and should be always only for calling an Azure Service.  So we always return `Response<T>` as per the guideline mandates :)

**Pull request closed without merging**